src/django_pg_migration_tools/management/commands/migrate_with_timeouts.py

```diff
@@ -28,32 +54,304 @@ def add_arguments(self, parser: base.CommandParser) -> None:
                 "must be specified."
             )
 
-        statement_timeout: datetime.timedelta | None = None
-        if statement_timeout_in_ms is not None:
-            statement_timeout = datetime.timedelta(
-                seconds=int(statement_timeout_in_ms / 1_000)
+        positive_fields = [
+            self.lock_timeout_max_retries,
+            self.lock_timeout_retry_exp,
+            self.lock_timeout_retry_max_wait,
+            self.lock_timeout_retry_min_wait,
+            self.statement_timeout_max_retries,
+            self.statement_timeout_retry_exp,
+            self.statement_timeout_retry_max_wait,
+            self.statement_timeout_retry_min_wait,
+        ]
+        for f in positive_fields:
+            # Validate for negative fields manually here instead of using a
+            # custom argparse check as Django does not run the validator prior
+            # calling handle(). At which point we don't have access to the
+            # parser instance for calling argparse's validation.
+            if isinstance(f, datetime.timedelta):
+                # time fields are expressed in milisseconds and so are printed
+                # back to the MC caller as such to avoid confusion.
+                ms = int(f.total_seconds() * 1_000)
+                if ms < 0:
+                    raise ValueError(f"{ms} is not a positive value.")
+            elif f < 0:
+                raise ValueError(f"{f} is not a positive value.")
+
+        if (
+            self.statement_timeout_retry_min_wait is not None
+            and self.statement_timeout_retry_max_wait is not None
+        ) and (
+            self.statement_timeout_retry_min_wait
+            > self.statement_timeout_retry_max_wait
+        ):
+            raise ValueError(
+                "statement-timeout-retry-min-wait-in-ms cannot be greater than "
+                "statement-timeout-retry-max-wait-in-ms."
             )
 
-        lock_timeout: datetime.timedelta | None = None
-        if lock_timeout_in_ms is not None:
-            lock_timeout = datetime.timedelta(seconds=int(lock_timeout_in_ms / 1_000))
+        if (
+            self.lock_timeout_retry_min_wait is not None
+            and self.lock_timeout_retry_max_wait is not None
+        ) and (self.lock_timeout_retry_min_wait > self.lock_timeout_retry_max_wait):
+            raise ValueError(
+                "lock-timeout-retry-min-wait-in-ms cannot be greater than "
+                "lock-timeout-retry-max-wait-in-ms."
+            )
 
-        with timeouts.apply_timeouts(
-            using=options["database"],
-            lock_timeout=lock_timeout,
-            statement_timeout=statement_timeout,
-        ):
-            super().handle(*args, **options)
+    def _migrate(self, *django_migrate_args: Any, **django_migrate_kwargs: Any) -> None:
+        migration_applied: bool = False
+        while self._can_migrate() and (migration_applied is False):
```

Comment:
```suggestion
        while self._can_migrate():
```
and then do a `return` instead of `migration_applied = True`

And then the raising of the exception does need the `if not migration_applied` guard cause the only time that would be reached would be if the migration didn't happen.